File: LeetCode/53_MaximumSubarray.java
// https://leetcode.com/problems/maximum-subarray/
// 2023年03月04日 13:49:52
// still no idea

class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0];
        for (int i = 1; i < nums.length; i++) {
            nums[i] = Math.max(nums[i - 1] + nums[i], nums[i]);
            maxSum = Math.max(maxSum, nums[i]);
        }
        return maxSum;
    }
}

File: LeetCode/53_MaximumSubarray.go
// https://leetcode.com/problems/maximum-subarray/
// 2023年03月25日 16:19:00
// still no idea
func maxSubArray(nums []int) int {
	maxVal := nums[0]
	for i := 1; i < len(nums); i++ {
		if nums[i-1]+nums[i] > nums[i] {
			nums[i] += nums[i-1]
		}
		if nums[i] > maxVal {
			maxVal = nums[i]
		}
	}
	return maxVal
}